### PR TITLE
specify seed in RayDataset shuffling

### PR DESCRIPTION
**Pull request by abidwael**
- Training the same model twice on the Ray backend yields different metric value results. This PR fixes that.
- It also modifies the `ludwig-ray` image to get the torch wheel from `https://download.pytorch.org/whl/cpu`

### CODE REVIEW

File: tests/ludwig/models/test_training_determinism.py [hunk 1]
```diff
@@ -0,0 +1,109 @@
...
+
+@pytest.mark.distributed
+def test_training_determinism_ray_backend(csv_filename, tmpdir):
+    experiment_output_1, experiment_output_2 = train_twice("ray", csv_filename, tmpdir)
+
+    eval_stats_1, train_stats_1, _, _ = experiment_output_1
+    eval_stats_2, train_stats_2, _, _ = experiment_output_2
+
+    assert json.dumps(eval_stats_1, cls=NumpyEncoder, sort_keys=True, indent=4) == json.dumps(
+        eval_stats_2, cls=NumpyEncoder, sort_keys=True, indent=4
```
Comment (justinxzhao) [inline tests/ludwig/models/test_training_determinism.py:36]: Do we need the extra json.dumps sorting step? I would expect pytest to be able to compare dicts (with differently sorted keys), as is.

Comment (abidwael) [reply tests/ludwig/models/test_training_determinism.py:36]: I resorted to `json.dumps` beacuse direct dict comparison fails in case there are nans.

Comment (justinxzhao) [reply tests/ludwig/models/test_training_determinism.py:36]: Hmm, patching your PR, I'm seeing that 

```
assert eval_stats_1 == eval_stats_2
assert train_stats_1 == train_stats_2
```

Passes.

Comment (justinxzhao) [reply tests/ludwig/models/test_training_determinism.py:36]: Nevermind, looks like it's not a consistent pass.

File: tests/ludwig/models/test_training_determinism.py [hunk 2]
```diff
@@ -0,0 +1,109 @@
...
+    )
+    assert json.dumps(train_stats_1, cls=NumpyEncoder, sort_keys=True, indent=4) == json.dumps(
```
Comment (ShreyaR) [inline tests/ludwig/models/test_training_determinism.py:38]: Are the training stats won't always be the exact same, even with full determinism? I'd assume that there's some decimal errors, in which case doing an `np.allclose` might be more appropriate instead of an `equals` assertion.

**Review by ShreyaR — COMMENTED** [review]
Lgtm! Can approve once the comment is addressed.

**Review by justinxzhao — COMMENTED** [review]
This looks like it should be broken up into separate PRs:
1. Update ludwig-ray docker image with torch CPU wheel.
2. Specify seed in RayDataset shuffling.
3. Customize logged message when early stopping is disabled.

While each change is small, we should get into the habit of making PRs cleanly self-encapsulated. Could you create separate PRs for the other changes?

File: tests/ludwig/models/test_training_determinism.py (hunk 2 above)
Comment (abidwael) [reply tests/ludwig/models/test_training_determinism.py]: Used `numpy.testing.assert_equal` which consider `np.nan` as if they were normal numbers.

File: tests/ludwig/models/test_training_determinism.py [hunk 3]
```diff
@@ -0,0 +1,99 @@
...
+
+
+@pytest.mark.distributed
+def test_training_determinism_ray_backend(csv_filename, tmpdir):
+    experiment_output_1, experiment_output_2 = train_twice("ray", csv_filename, tmpdir)
+
+    eval_stats_1, train_stats_1, _, _ = experiment_output_1
+    eval_stats_2, train_stats_2, _, _ = experiment_output_2
+
+    np.testing.assert_equal(eval_stats_1, eval_stats_1)
```
Comment (justinxzhao) [inline tests/ludwig/models/test_training_determinism.py:34]: Nice!

**Review by justinxzhao — APPROVED** [review]
